src/pages/Fox/hooks/useFoxPageContext.tsx

```diff
@@ -0,0 +1,64 @@
+import { type AccountId, type AssetId, fromAssetId } from '@shapeshiftoss/caip'
+import React, { createContext, useContext, useMemo, useState } from 'react'
+import {
+  selectAccountIdByAccountNumberAndChainId,
+  selectAccountNumberByAccountId,
+} from 'state/slices/portfolioSlice/selectors'
+import { useAppSelector } from 'state/store'
+
+type FoxPageContextType = {
+  selectedAssetAccountId: AccountId | undefined
+  assetId: AssetId
+  setAssetAccountId: React.Dispatch<React.SetStateAction<AccountId | undefined>>
+  assetAccountNumber: number | undefined
+}
+
+const FoxPageContext = createContext<FoxPageContextType | undefined>(undefined)
+
+export const FoxPageProvider: React.FC<React.PropsWithChildren<{ assetId: AssetId }>> = ({
+  children,
+  assetId,
+}) => {
+  const [assetAccountId, setAssetAccountId] = useState<AccountId | undefined>()
+
+  const filter = useMemo(
+    () => (assetAccountId && assetId ? { assetId, accountId: assetAccountId } : undefined),
+    [assetAccountId, assetId],
+  )
+
+  const assetAccountNumber = useAppSelector(state =>
+    filter ? selectAccountNumberByAccountId(state, filter) : undefined,
+  )
+
+  const accountIdsByAccountNumberAndChainId = useAppSelector(
+    selectAccountIdByAccountNumberAndChainId,
+  )
+
+  const selectedAssetAccountId = useMemo(() => {
+    if (!(filter && assetAccountNumber !== undefined)) return
+    const accountNumberAccountIds = accountIdsByAccountNumberAndChainId[assetAccountNumber]
+    const matchingAccountId = accountNumberAccountIds?.[fromAssetId(assetId).chainId]
+    return matchingAccountId
+  }, [accountIdsByAccountNumberAndChainId, filter, assetId, assetAccountNumber])
```

Comment:
Q: this is copied from rfox but do we need this here? as far as this section is concerned doesn't look like we do, and RFOX should be able to leverage RFOX context with the notion of matching asset AccountId there

Reply:
We do, we need to select the matching fox account to display it's balances

rFOX section will leverage `assetAccountNumber` to select it's arb account

Reply:
@NeOMakinG looks like naming is still off, which is why the confusion may be stemming from:

https://github.com/shapeshift/web/blob/c514772f6ee449d73940b872a144ecaf31d8a826/src/pages/Fox/hooks/useFoxPageContext.tsx#L35-L49

As you can see, we assign `assetAccountId` to `selectedAssetAccountId`, which is wrong, since there is no notion of selected asset.

As mentioned in the initial comment, there will be a notion of selected asset in the future, and that should be handled by the RFOX context.